code/modules/power/port_gen.dm

```diff
@@ -312,78 +312,72 @@
-	if(IsBroken())
-		return
+	return attack_hand(user)
 
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/power/port_gen/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		ui = new(user, src, ui_key, "pacman.tmpl", src.name, 500, 560)
+		ui = new(user, src, ui_key, "Pacman", name, 500, 260)
 		ui.open()
-		ui.set_auto_update(1)
 
-/obj/machinery/power/port_gen/pacman/ui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
-	var/data[0]
+/obj/machinery/power/port_gen/pacman/tgui_data(mob/user)
+	var/list/data = list()
+	var/sheetVolume = 1000 //cm3
 
 	data["active"] = active
 	if(istype(user, /mob/living/silicon/ai))
-		data["is_ai"] = 1
+		data["is_ai"] = TRUE
 	else if(istype(user, /mob/living/silicon/robot) && !Adjacent(user))
-		data["is_ai"] = 1
+		data["is_ai"] = TRUE
 	else
-		data["is_ai"] = 0
+		data["is_ai"] = FALSE
 
+	data["anchored"] = anchored
+	data["broken"] = IsBroken()
 	data["output_set"] = power_output
 	data["output_max"] = max_power_output
 	data["output_safe"] = max_safe_output
-	data["output_watts"] = power_output * power_gen
-	data["temperature_current"] = src.temperature
-	data["temperature_max"] = src.max_temperature
-	data["temperature_overheat"] = overheating
-	// 1 sheet = 1000cm3?
-	data["fuel_stored"] = round((sheets * 1000) + (sheet_left * 1000))
-	data["fuel_capacity"] = round(max_sheets * 1000, 0.1)
-	data["fuel_usage"] = active ? round((power_output / time_per_sheet) * 1000) : 0
+	data["power_gen"] = power_gen
+	data["tmp_current"] = temperature
+	data["tmp_max"] = max_temperature
+	data["tmp_overheat"] = overheating
+	data["fuel_stored"] = round((sheets * sheetVolume) + (sheet_left * sheetVolume))
+	data["fuel_cap"] = round(max_sheets * sheetVolume, 0.1)
+	data["fuel_usage"] = active ? round((power_output / time_per_sheet) * sheetVolume) : 0
 	data["fuel_type"] = sheet_name
 
 	return data
 
-/obj/machinery/power/port_gen/pacman/Topic(href, href_list)
+/obj/machinery/power/port_gen/pacman/tgui_act(action, params)
 	if(..())
 		return
 
-	src.add_fingerprint(usr)
-	if(href_list["action"])
-		if(href_list["action"] == "enable")
-			if(!active && HasFuel() && !IsBroken())
-				active = 1
-				update_icon()
-		if(href_list["action"] == "disable")
-			if(active)
-				active = 0
-				update_icon()
-		if(href_list["action"] == "eject")
-			if(!active)
-				DropFuel()
-		if(href_list["action"] == "lower_power")
-			if(power_output > 1)
-				power_output--
-		if(href_list["action"] == "higher_power")
-			if(power_output < max_power_output || (emagged && power_output < round(max_power_output*2.5)))
-				power_output++
-
-	SSnanoui.update_uis(src)
+	add_fingerprint(usr)
+
+	. = TRUE
+
+	switch(action)
+		if("toggle_power")
+			if(!HasFuel())
+				atom_say("Out of fuel. ")
```

Comment:
Space at the end.
Also this should be unreachable - "on" button should be disabled (and an 'no fuel' shown in interface) if generator has no fuel.

Reply:
done